[Lens] Color in dimension trigger

flash1293 opened the pull request:
Based on https://github.com/elastic/kibana/pull/79851

Fixes https://github.com/elastic/kibana/issues/68405

This unfinished PR adds color indicators for dimension in xy chart.

Please don't review in depth yet, this PR is not cleaned up yet.

## Feature

The indicators are shown when possible. If the color can't be determined (because "break down by" is used), the "disabled" icon is shown. For dimensions containing a color palette picker, the currently picked palette is shown in the dimension picker.

<img width="636" alt="Screenshot 2020-11-05 at 10 43 41" src="https://user-images.githubusercontent.com/1508364/98224574-e9314900-1f53-11eb-8b1f-a6dbf37af5d5.png">
<img width="279" alt="Screenshot 2020-11-05 at 10 43 22" src="https://user-images.githubusercontent.com/1508364/98224589-ec2c3980-1f53-11eb-9fef-f21e7c6c38e6.png">
<img width="500" alt="Screenshot 2020-11-05 at 10 42 55" src="https://user-images.githubusercontent.com/1508364/98224591-ecc4d000-1f53-11eb-8c0e-aaa2d2030ca6.png">

The color indicator does not contain a color picker itself - to change the color you still have to open the dimension popover. As soon as the flyout dimension config PR lands, I think this is justifiable as it's just a single click away.

## Implementation

This PR extends the chart config like this: https://github.com/elastic/kibana/pull/76871/files#diff-c46757b63dd572e7cff7c0aad72c3ec135b9969c39716d4db9df551fa962ce34R347-R361

When passing the accessors to the frame, it's possible to specify a color icon as well. Another option would be to provide a `renderColorIndicator` or something along those lines (maybe making it more than generic than being about colors) and using a separate `NativeRenderer`. This would give the visualization more flexibility - I thought it would add unnecessary complexity in this specific situation, but I don't hold this opinion strongly.

The logic to determine the current color of a dimension (https://github.com/elastic/kibana/pull/76871/files#diff-5dcc03dab8d34631c6ed65b6b38f8b97eed0394699532c6516111c735a8e30bfR177-R220) works like this:
* Check whether currently rendered data is available - if not, don't show an indicator
* Check whether current layer has a break down by dimension - if yes, show the disabled icon
* Check whether there is an override color - if yes, show that
* Use the same color assignment logic as the actual chart to determine the current series color

Quick note about layout: To make the color indicator clickable like the dimension trigger rendered by the datasource, the `EuiLink` component rendering the button is moved out of the datasource into the frame - this means the datasource only has to render the text instead of having to handle the logic to actually open the popover: https://github.com/elastic/kibana/pull/76871/files#diff-9d5c4c19a499161633fc439fb1b86a6105bf7b2af2cdf638c52cec4f2a3c4f3cL93

Other changes:

* Make the color assignment logic resilient to malformed data and make it work with both expression level types and state level types: x-pack/plugins/lens/public/xy_visualization/color_assignment.ts
* Fix a potential bug in color assignment logic where the default palette was not picked correctly https://github.com/elastic/kibana/pull/76871/files#diff-1ca248a5e9aa96a613a49b87904efc6ce9e931e496f3f4dd7ccd15aa580d8e61R29
* Extract column label to accessor map generation out of the expression function to be able to use it for coloring logic as well: https://github.com/elastic/kibana/pull/76871/files#diff-95f4d1c35bb73bd6c00ce53520dcdb9a3ac2db7707fae067542fd218e9f9a36dR42

flash1293 commented:
Ping @AlonaNadler @cchaos @mbondyra @dej611 @wylieconlon - this is an unpolished version of the color indicator in the dimension trigger. In the bottom of the description I noted some open question - feedback is welcome.

mbondyra commented:
@flash1293 I really like this PR! I agree that not showing the color after 'break-down by' layer is not ideal, but at least we're not showing the incorrect information and it's quite an edge case, given the fact that not a lot of users use layers at the moment.

dej611 commented:
PR looks good! 👍 
I have only some (minor) perplexity on the "unknown" color icon: it is consistent with the rest of the UI, but at the same time I do not read it as "unknown" color, rather as "no color".
Have you considered not showing anything at all in its place?

flash1293 commented:
That's a valid point @dej611 - We are using the same icon for "auto" in the popover:
<img width="546" alt="Screenshot 2020-09-08 at 10 23 15" src="https://user-images.githubusercontent.com/1508364/92451683-5aa38400-f1bd-11ea-9122-ec56af46fb4d.png">

But you are right that this is probably the first place a user will see the icon, so they have no good way of learning it means "auto". Maybe a tooltip would help? Something like `Color will be defined by "break down by" dimension`.

flash1293 commented:
@AlonaNadler I updated the PR, it's now also using the flyout.

AlonaNadler commented:
Looks good @flash1293 
few things:
- Can you show the color instead of writing auto. The reason is that people scan the form since its not colorful its harder to distinguish, also it doesn't help - auto still has color


> Would it be OK to only show the indicator for overridden colors instead of all dimensions?

the pr currently shows all dimensions and I think its a good call

> Is it OK if it's not possible to pick the color directly from the indicator?
yes, ideally the color should be selected in the legend to allow users to change it in dashboards. We will need to address it in the future


> Is it OK to solve the color palette case separately?
Not sure what it means 

> Is the edge case about a bad icon choice in non-broken down layers after a broken down layer relevant?


Not sure what it means 

> For pie charts, should the first "slice by" dimension also have the "color by" icon? It would be more correct, however I'm not sure whether it helps as you can't configure anything about it

Yes I agree it's not ideal. Ideally, you can select a pallete for the pie chart and also decide to overide a slice and choose its color. At the minimum we need to allow to select the color palette

cchaos commented:
_Possibly controversial question_

**Can we delay this PR til 7.11?** 

I have a lot of feedback on it currently that will probably require discussion. I also see current discussions that end with  “can’t happen right now” which I then find this feature to be incomplete and most likely more confusing.

There is a "best" way to handle indicating color and I think it hinges on people being able to customize more than just the Y-axis. It also hinges on getting right of "Auto" color everywhere. The use of the square with a slash was supposed to be temporary and this PR buys **even more** into the concept.

The designs from #68405 are now several months old and I'm not sure this concept is going to work as well with the breakdown by's especially with multiple layers. So I'd like to pause this one til after FF.

flash1293 commented:
Happy to delay this, @cchaos - I fully understand your concerns. We can discuss this as part of color palettes https://github.com/elastic/kibana/issues/53661

@AlonaNadler would you object going through another iteration on this and finding a better solution with to cost of pushing it out a version?

flash1293 commented:
I'm going to park this until https://github.com/elastic/kibana/pull/75309 is completed - then I can solve it cleanly from the start

flash1293 commented:
@elasticmachine merge upstream

kibanamachine commented:
merge conflict between base and head

flash1293 commented:
Jenkins, test this

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/)
* Commit: 61e4849ef78cdf598cfef1f16def49ceba614181

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/7904ee0ceb7c9219d5fe3816e90239fefb8945a9) | [after](https://github.com/elastic/kibana/commit/61e4849ef78cdf598cfef1f16def49ceba614181) | diff |
| --- | --- | --- | --- |
| `lens` | 1.0MB | 1.0MB | +5.7KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/7904ee0ceb7c9219d5fe3816e90239fefb8945a9) | [after](https://github.com/elastic/kibana/commit/61e4849ef78cdf598cfef1f16def49ceba614181) | diff |
| --- | --- | --- | --- |
| `lens` | 50.3KB | 52.1KB | +1.7KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/","number":"86508","commit":"61e4849ef78cdf598cfef1f16def49ceba614181"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

flash1293 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86644/)
* Commit: d32cb665fb34c4d1fbc93a792df207320afe63e4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/8560b2dcc00f3992b43b75daf844ce7834678ac8) | [after](https://github.com/elastic/kibana/commit/d32cb665fb34c4d1fbc93a792df207320afe63e4) | diff |
| --- | --- | --- | --- |
| `lens` | 1.0MB | 1.0MB | +5.7KB |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/8560b2dcc00f3992b43b75daf844ce7834678ac8) | [after](https://github.com/elastic/kibana/commit/d32cb665fb34c4d1fbc93a792df207320afe63e4) | diff |
| --- | --- | --- | --- |
| `lens` | 50.3KB | 52.1KB | +1.7KB |

### History
* :green_heart: [Build #86508](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/) succeeded 61e4849ef78cdf598cfef1f16def49ceba614181

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/","number":"86508","commit":"61e4849ef78cdf598cfef1f16def49ceba614181"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86644/","number":"86644","commit":"d32cb665fb34c4d1fbc93a792df207320afe63e4"}]}
PIPELINE-->

mbondyra commented on x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.tsx:
```diff
@@ -297,26 +303,66 @@ export function LayerPanel(
-                          />
+                          >
+                            <EuiFlexGroup gutterSize="none" alignItems="center">
+                              {typeof accessorConfig !== 'string' &&
```
Could we move this logic to separate component that takes `triggerIcon` and `color` props, eg. <ColorIndicator />?

mbondyra commented on x-pack/plugins/lens/public/types.ts:
```diff
@@ -345,12 +344,21 @@ export type VisualizationDimensionEditorProps<T = unknown> = VisualizationConfig
 };
 
+export type AccessorConfig =
+  | string
```
I wonder if it wouldn't be better to not support this type anymore and convert all the accessors to the format below (`{columnId:... } `). We already introduce 6 checks in layer_panel only to check if `typeof accessorConfig !== 'string'`...

mbondyra commented on x-pack/plugins/lens/public/xy_visualization/visualization.tsx:
```diff
@@ -166,9 +170,55 @@ export const getXyVisualization = ({
     const datasource = frame.datasourceLayers[layer.layerId];
 
-    const sortedAccessors = getSortedAccessors(datasource, layer);
+    const sortedAccessors: string[] = getSortedAccessors(datasource, layer);
```
Could we extract the logic of assigning `mappedAccessors` to function for readability?

mbondyra commented:
What an amazing PR 😻 

From small things I noticed:

This PR changed slightly the styles for focused dimension trigger: 
![image](https://user-images.githubusercontent.com/4283304/98671768-10718700-2355-11eb-932d-5d2863e53fed.png)


##### Opinion land:
I wonder if in this case we want to display the color placeholder (the little square) for yAxis at all... I feel like skipping it would be less confusing. Or maybe we could add a tooltip of what it means? 
![image](https://user-images.githubusercontent.com/4283304/98671987-5d555d80-2355-11eb-89bd-6a5b6b1f4df4.png)

flash1293 commented:
Pinging @MichaelMarcialis for the style of color palettes inside of dimension triggers and Martas suggestion here: https://github.com/elastic/kibana/pull/76871#issuecomment-724713606

MichaelMarcialis commented:
This is looking great, @flash1293. While comparing your screenshots to @cchaos's latest (I think) [mockups for chart coloring](https://www.figma.com/file/CJfejj3KoCnJqx58wzfCvf/%E2%AD%90Flows?node-id=4494%3A3354), I noticed that there were a few places that this PR deviates from those designs, including:

- The `brush` icon appears to be used, where the mockups use `color`.
- The `stopSlash` icon appears to be used, where the latest mockups appear to use no icon when axis color indicators are removed due to a breakdown palette being added.
- The `brush` icon uses `$euiLinkColor`, where the mockups use the default text color.
- The color bar placement here is flush with the left/bottom/right of its container, but is padded within its container in the mockups.

Was there a reason for these deviations? Am I looking at an older mockup, or is this just the first part of a multi-PR effort (as your initial comment appears to indicate)? Thanks!

![image](https://user-images.githubusercontent.com/3884767/98739819-0b71ff00-2378-11eb-9848-678747536ba8.png)

MichaelMarcialis commented:
> This PR changed slightly the styles for focused dimension trigger:
> <img alt="" height="200" src="https://user-images.githubusercontent.com/4283304/98671768-10718700-2355-11eb-932d-5d2863e53fed.png">

Is there any way we can fix this so that the focus states updated as part of @mbondyra's [recent reordering PR](https://github.com/elastic/kibana/pull/80547) can remain intact? Figma examples of what was she implemented can be found here: https://www.figma.com/file/q61JmGHmcammPfkxupNj1g/Intradimensional-Reordering-Version-1-In-Progress?node-id=1%3A2

> I wonder if in the case below we want to display the the little square logo for yAxis at all... I feel like skipping it would be less confusing. Or maybe we could add a tooltip to explain what it means?
> 
> <img alt="" height="200" src="https://user-images.githubusercontent.com/4283304/98671987-5d555d80-2355-11eb-89bd-6a5b6b1f4df4.png">

Per my earlier comment, @cchaos's most recent mockups appear to omit this `stopSlash` icon altogether (unless I'm looking at an outdated file). Personally, after having seen it both ways, I tend to like the inclusion of the icon because it:

1. Provides consistency between dimensional items, in terms of having a preceding icon, having the text aligned the same across items, etc. 
2. Is potentially helpful for users to quickly identify existing or possible areas of color control.

Regarding applying a tooltip to these `stopSlash` icons, I'm a bit hesitant to recommend that direction, as there's already a lot interaction already attached to each dimensional item. Most importantly, I don't wish to potentially interfere with the drag-and-drop (i.e. hovering the icon opens a tooltip, which inadvertently prevents a portion of the item from being draggable). I wonder if simply using a lower contrast icon color (such as `$euiColorMediumShade`) would put less of a spotlight on it and better indicate that it's a swatch chip in a disabled/empty state. Perhaps we could also explain further in the item's color field in the configuration flyout that it's being overridden. Thoughts?

flash1293 commented:
Thanks for the reviews. I tried to address all of the things, this is the summary:
* Code:
   * Moved color logic into separate components/functions
   * Removed the ability to specify an accessor config as string (you are right Marta, not really necessary)
* Behavior:
   * Using text color for `brush`
   * Fixing focus style
   * Using subdued color for `stopSlash`

@MichaelMarcialis

> Personally, after having seen it both ways, I tend to like the inclusion of the icon

 I left it in with a little less prominent color. No strong opinion here.

About the differences not addressed above:

> The brush icon appears to be used, where the mockups use color.

I didn't find the right icon, maybe it's not released yet? If I'm using `<EuiIcon type="color"`, I don't get an icon at all. Am I looking in the wrong place?

> The color bar placement here is flush with the left/bottom/right of its container, but is padded within its container in the mockups.

I did that as a quick and dirty solution, but then I ended up liking it because it's not adding to the vertical space. If you think it's not the right thing I can move to the style in Carolines mock.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87397/)
* Commit: 188ed0870d19cdac594d32d54a68b3cc2e3004e1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/01daf28e75ab7507d09cd82aa145376b2404b613) | [after](https://github.com/elastic/kibana/commit/188ed0870d19cdac594d32d54a68b3cc2e3004e1) | diff |
| --- | --- | --- | --- |
| `lens` | 432 | 434 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/01daf28e75ab7507d09cd82aa145376b2404b613) | [after](https://github.com/elastic/kibana/commit/188ed0870d19cdac594d32d54a68b3cc2e3004e1) | diff |
| --- | --- | --- | --- |
| `lens` | 914.6KB | 921.4KB | +6.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/01daf28e75ab7507d09cd82aa145376b2404b613) | [after](https://github.com/elastic/kibana/commit/188ed0870d19cdac594d32d54a68b3cc2e3004e1) | diff |
| --- | --- | --- | --- |
| `lens` | 50.6KB | 52.3KB | +1.7KB |

### History
* :green_heart: [Build #86644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86644/) succeeded d32cb665fb34c4d1fbc93a792df207320afe63e4
* :green_heart: [Build #86508](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/) succeeded 61e4849ef78cdf598cfef1f16def49ceba614181

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/","number":"86508","commit":"61e4849ef78cdf598cfef1f16def49ceba614181"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86644/","number":"86644","commit":"d32cb665fb34c4d1fbc93a792df207320afe63e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87397/","number":"87397","commit":"188ed0870d19cdac594d32d54a68b3cc2e3004e1"}]}
PIPELINE-->

MichaelMarcialis commented on x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/palette_indicator.tsx:
```diff
@@ -0,0 +1,27 @@
+export function PaletteIndicator({ accessorConfig }: { accessorConfig: AccessorConfig }) {
+  if (accessorConfig.triggerIcon !== 'colorBy' || !accessorConfig.palette) return null;
+  return (
+    <EuiFlexGroup className="lnsLayerPanel__paletteContainer" gutterSize="none" alignItems="center">
+      {accessorConfig.palette.map((color) => (
+        <EuiFlexItem
+          key={color}
+          className="lnsLayerPanel__paletteColor"
+          grow={true}
+          style={{
+            backgroundColor: color,
+          }}
+        />
+      ))}
+    </EuiFlexGroup>
```
Rather than rolling out a custom color palette component, would it be better to use `EuiColorPaletteDisplay` instead (and override as necessary if needed)?

https://elastic.github.io/eui/#/forms/color-selection

flash1293 replied:
This component is very new - the version supporting it is not available in Kibana yet. I'm fine with using it once it becomes available, but https://github.com/elastic/kibana/pull/82730 has to get merged first.

MichaelMarcialis replied:
Sounds good. Should I open an issue as a reminder? Or is that unnecessary?

flash1293 replied:
Don't bother, I will take care of it.

MichaelMarcialis commented on x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_panel.tsx:
```diff
@@ -86,11 +86,11 @@ export const IndexPatternDimensionTriggerComponent = function IndexPatternDimens
         anchorClassName="eui-displayBlock"
       >
-        <EuiLink
+        <EuiText
```
There are a few questions/concerns I have due to the change from `EuiLink` to `EuiText`:

- Since this element is no longer focusable, the styles applied by @mbondyra in a previous PR no longer set the focus styles (light blue background color behind text) appropriately for the child `.lnsLayerPanel__triggerLinkLabel` element. Can this be corrected?
- I believe this `aria-label` now no longer applies to this element. Can we move to the appropriate location?
- Should the `lnsLayerPanel__triggerLink` class name be changed, since it's no longer a link?

MichaelMarcialis reviewed (commented):
Hey, @flash1293. Took a deeper look at the code and left a few comments for your review.

flash1293 commented:
Thanks for the review, @MichaelMarcialis 

These are the changes I made:
* Use eui icon for solid color indicators as well
* Set aria labels for them
* Didn't change palette to the EUI implementation as this component is not available in Kibana yet
* Fixed focus styles (I hope :) ) - I moved the focus style logic to the right class
* Renamed `lnsLayerPanel__triggerLink` to `lnsLayerPanel__triggerText` as the link part is now handled by the frame in `lnsLayerPanel__dimensionLink`- move aria label / title to the button element again as well

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87931/)
* Commit: 4e0a21f8028878391c074b16d521a7fc3c9857ff

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/253495b6d76fd29f57f3a93f4eac99dd64cc88e7) | [after](https://github.com/elastic/kibana/commit/4e0a21f8028878391c074b16d521a7fc3c9857ff) | diff |
| --- | --- | --- | --- |
| `lens` | 433 | 435 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/253495b6d76fd29f57f3a93f4eac99dd64cc88e7) | [after](https://github.com/elastic/kibana/commit/4e0a21f8028878391c074b16d521a7fc3c9857ff) | diff |
| --- | --- | --- | --- |
| `lens` | 917.9KB | 924.7KB | +6.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/253495b6d76fd29f57f3a93f4eac99dd64cc88e7) | [after](https://github.com/elastic/kibana/commit/4e0a21f8028878391c074b16d521a7fc3c9857ff) | diff |
| --- | --- | --- | --- |
| `lens` | 50.7KB | 52.4KB | +1.7KB |

### History
* :green_heart: [Build #87397](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87397/) succeeded 188ed0870d19cdac594d32d54a68b3cc2e3004e1
* :green_heart: [Build #86644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86644/) succeeded d32cb665fb34c4d1fbc93a792df207320afe63e4
* :green_heart: [Build #86508](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/) succeeded 61e4849ef78cdf598cfef1f16def49ceba614181

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86508/","number":"86508","commit":"61e4849ef78cdf598cfef1f16def49ceba614181"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/86644/","number":"86644","commit":"d32cb665fb34c4d1fbc93a792df207320afe63e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87397/","number":"87397","commit":"188ed0870d19cdac594d32d54a68b3cc2e3004e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/87931/","number":"87931","commit":"4e0a21f8028878391c074b16d521a7fc3c9857ff"}]}
PIPELINE-->

MichaelMarcialis reviewed (approved):
Thank you for addressing those suggested changes. This LGTM.

MichaelMarcialis commented:
> > The color bar placement here is flush with the left/bottom/right of its container, but is padded within its container in the mockups.
> 
> I did that as a quick and dirty solution, but then I ended up liking it because it's not adding to the vertical space, preventing scrolling for a little bit longer. If you think it's not the right thing I can move to the style in Carolines mock.

Personally, I like how you've positioned the palette here and can appreciate the vertical real estate savings. Was just checking to see if there was a different reason for the deviation. I say we keep it positioned as you currently have it.

mbondyra reviewed (approved):
tested on Safari, works as expected. Code LGTM 🟢

flash1293 commented:
Merging now, I will replace the custom palette code with the EUI component once it becomes available
